### ISSUE TITLE
Fails on compile time for WASM target.

### ISSUE THREAD

**Issue by webbertakken** [getsentry/sentry-rust#513]
### Environment

Latest rust, latest bevy, sentry 0.27.0.

### Steps to Reproduce

Use [this template](https://github.com/NiklasEi/bevy_game_template) and build for Wasm using `trunk serve`

Add with default dependencies:

sentry = { version = "0.27.0" }

Run `trunk serve` and notice that during compile time it's giving an error exactly like [this one](https://github.com/svartalf/hostname/issues/16#issue-1111904926).

### Expected Result

I would expect sentry package to work out of the box when using default features.

The package `hostname` will not be solving the issue because they prefer it that way. (see [this comment](https://github.com/svartalf/hostname/pull/15#issuecomment-1009042477) from the owner).

So in order for that to happen, the `hostname` package should probably become an optional dependency which is not applied for WASM builds.

### Actual Result

We have to customise features like so (I did not end up even getting it to work at all)

```
sentry = { version = "0.27.0", default-features = false, features = [
  "backtrace",
#  "contexts", # Will not work for WASM target
  "panic",
#  "transport", # no `Proxy` in the root
] }
```

- Enabling contexts, will give said error
- Keeping `transport` disabled will panic during runtime:

![image](https://user-images.githubusercontent.com/20756439/200144200-4741e6ac-b092-4f45-9fdb-77b3bbabfbde.png)

- Enabling transport will give the following errors:

![image](https://user-images.githubusercontent.com/20756439/200144171-42e02fa1-4b31-4bad-979a-98ec405b1d65.png)

Any advice on how to get this working?

**Comment by omarabid** [comment]
I don't think this will be working anytime soon. Apart of the dependencies that do not support WASM target, all types should be `?Send` since WASM is a single threaded environment.

**Comment by hut8** [comment]
Seems like a duplicate of https://github.com/getsentry/sentry-rust/issues/226

I wish we could get this working, or at least have a very clear guide on how to integrate with a build based on wasmpack.
